src/main/java/com/google/sps/servlets/TagServlet.java

```diff
@@ -122,4 +136,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.getWriter().println(json);
   }
 
+  public Long strToLong(String str) {
+    if (str == null) {
+      return null;
+    } else {
+      return Long.parseLong(str);
+    }
+  }
+
```

Comment:
This function is repeated in more than 1 places. Consider defining a util class to hold such functions.